### ISSUE TITLE
OpenVDB v3.2.0 Python Module Fails to Compile with Python v3.x 

### ISSUE THREAD

**Issue by dracwyrm** [dreamworksanimation/openvdb#68]
Hi,

Using the source from master, at the date in the title, the python module fails with this message:

```
In file included from /usr/include/sched.h:28:0,
                 from /usr/include/pthread.h:23,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/x86_64-pc-linux-gnu/bits/gthr-default.h:35,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/x86_64-pc-linux-gnu/bits/gthr.h:148,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/ext/atomicity.h:35,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/bits/basic_string.h:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/string:52,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/include/g++-v5/stdexcept:39,
                 from /usr/include/boost/function/function_base.hpp:14,
                 from /usr/include/boost/function/detail/prologue.hpp:17,
                 from /usr/include/boost/function/function_template.hpp:13,
                 from /usr/include/boost/function/detail/maybe_include.hpp:13,
                 from /usr/include/boost/function/function0.hpp:11,
                 from /usr/include/boost/python/errors.hpp:13,
                 from /usr/include/boost/python/handle.hpp:11,
                 from /usr/include/boost/python/args_fwd.hpp:10,
                 from /usr/include/boost/python/args.hpp:10,
                 from /usr/include/boost/python.hpp:11,
                 from python/pyOpenVDBModule.cc:33:
python/pyOpenVDBModule.cc: In function ‘void init_module_pyopenvdb()’:
python/pyOpenVDBModule.cc:595:5: error: return-statement with a value, in function returning 'void' [-fpermissive]
     import_array();
```

However, Python v2.7.x will compile the module. If you need more details, please let me know.

Thanks,
Jon

Appendix: The compile settings:
make -j2 python rpath=no shared=yes LIBOPENVDB_RPATH= DESTDIR=/var/tmp/portage/media-gfx/openvdb-3.2.0_pre20160604/work/install/usr HFS=/usr HT=/usr HDSO=/usr/lib64 CPPUNIT_INCL_DIR=/usr/include/cppunit CPPUNIT_LIB_DIR=/usr/lib64 LOG4CPLUS_INCL_DIR=/usr/include/log4cplus LOG4CPLUS_LIB_DIR=/usr/lib64 GLFW_INCL_DIR=/usr/lib64 GLFW_LIB_DIR=/usr/lib64 BLOSC_INCL_DIR=/usr/include BLOSC_LIB_DIR=/usr/lib64 DOXYGEN= PYTHON_VERSION=3.4 PYTHON_INCL_DIR=/usr/include/python3.4m PYCONFIG_INCL_DIR=/usr/include/python3.4m PYTHON_LIB_DIR=/usr/lib64/libpython3.4m.so PYTHON_LIB=-lpython3.4m PYTHON_INSTALL_INCL_DIR=/var/tmp/portage/media-gfx/openvdb-3.2.0_pre20160604/work/install/usr/include/python3.4m PYTHON_INSTALL_LIB_DIR=/var/tmp/portage/media-gfx/openvdb-3.2.0_pre20160604/work/install/usr/lib64/python3.4/site-packages NUMPY_INCL_DIR=/usr/lib64/python3.4/site-packages/numpy/core/include/numpy BOOST_PYTHON_LIB_DIR=/usr/lib64 BOOST_PYTHON_LIB=-lboost_python-3.4 EPYDOC= 
Building python/pyFloatGrid.o because of pyFloatGrid.cc and others
g++ -c -march=native -O2 -w -pipe -pthread -O3 -DNDEBUG -I . -I .. -isystem /usr/include -isystem /usr/include -isystem /usr/include -isystem /usr/include -isystem /usr/include/log4cplus -DOPENVDB_USE_BLOSC -DOPENVDB_USE_LOG4CPLUS -DOPENVDB_USE_GLFW_3 -I . -fPIC -isystem python -isystem /usr/include/python3.4m -isystem /usr/include/python3.4m -isystem /usr/lib64/python3.4/site-packages/numpy/core/include/numpy -DPY_OPENVDB_USE_NUMPY -o python/pyFloatGrid.o python/pyFloatGrid.cc
Building python/pyIntGrid.o because of pyIntGrid.cc and others
g++ -c -march=native -O2 -w -pipe -pthread -O3 -DNDEBUG -I . -I .. -isystem /usr/include -isystem /usr/include -isystem /usr/include -isystem /usr/include -isystem /usr/include/log4cplus -DOPENVDB_USE_BLOSC -DOPENVDB_USE_LOG4CPLUS -DOPENVDB_USE_GLFW_3 -I . -fPIC -isystem python -isystem /usr/include/python3.4m -isystem /usr/include/python3.4m -isystem /usr/lib64/python3.4/site-packages/numpy/core/include/numpy -DPY_OPENVDB_USE_NUMPY -o python/pyIntGrid.o python/pyIntGrid.cc

**Comment by dracwyrm** [comment]
I've been looking into this and it seems that NumPy has changed the return value via this code:
```
#if PY_VERSION_HEX >= 0x03000000
#define NUMPY_IMPORT_ARRAY_RETVAL NULL
#else
#define NUMPY_IMPORT_ARRAY_RETVAL
#endif
```
So in Python v3+ it will return a NULL and in less than v3, it returns nothing.

Some people use code like this call NumPy:
```
#if PY_MAJOR_VERSION >= 3
int
#else
void
#endif
init_numpy()
{
	import_array();
}
```

**Comment by dracwyrm** [comment]
See Pull Request #69

**Comment by dracwyrm** [comment]
Updated PR with a much cleaner way of doing this after I did a bit more research. Please review the PR.

**Comment by dracwyrm** [comment]
This bug still exists in the new 3.2 release.

**Comment by 0joshuaolson1** [comment]
The team hasn't merged that PR. Maybe you need to [sign](https://github.com/dreamworksanimation/openvdb#contributing) the [CLA](http://www.openvdb.org/documentation/doxygen/faq.html#sWhatCLA)?

**Comment by dracwyrm** [comment]
This is fixed in 4.4.0.
